### PR TITLE
Change announcement type on completion of syncing a historical block

### LINKED ISSUE #3556: Send DeployProcessed events to event stream server for historical blocks
We want to replace `BlockSynchronizerAnnouncement::CompletedBlock` announcement with a hot block announcement, where the hot block state is set up so that only the following events are emitted:
* `deploy_buffer::Event::Block`
* `consensus::Event::BlockAdded`
* `event_stream_server::Event::BlockAdded`
* `event_stream_server::Event::DeployProcessed` for each deploy (this will mean the BlockSynchronizer will need to read the corresponding execution results from storage, or retain them in memory)

We also have some nitpicks to resolve:
* rename hot block to meta block
* rename `State::new_synced()` to `new_already_stored()`
* remove `State::immediate_switch_block_for_current_protocol_version` and rename `State::new_immediate_switch()` to `new_should_not_gossip()`

### PR DESCRIPTION
**Pull request by Fraser999**
This PR includes the following changes (in three separate commits):
* renames hot block to meta block
* `BlockSynchronizer` uses a `MetaBlockAnnouncement` rather than a `BlockSynchronizerAnnouncement::CompletedBlock` when completing sync for a historical block.  This now causes the event stream server to emit execution results for historical blocks
* various renames and tweaks to meta block types

I verified via nctl that execution results are emitted for historical blocks by a joining node.

Closes #3556.

### CODE REVIEW

**Review by EdHastingsCasperAssociation — APPROVED** [review]

File: node/src/components/storage.rs [hunk 1]
```diff
@@ -2497,13 +2503,55 @@ impl Storage {
...
             }
         }
+        Ok(Some(execution_results))
+    }
 
+    #[allow(clippy::type_complexity)]
+    fn read_execution_results(
+        &self,
+        block_hash: &BlockHash,
+    ) -> Result<Option<Vec<(DeployHash, DeployHeader, ExecutionResult)>>, FatalStorageError> {
```
Comment (rafal-ch) [inline node/src/components/storage.rs:2513]: :see_no_evil:
;-)

File: node/src/types/block/meta_block/state.rs [hunk 2]
```diff
@@ -50,23 +49,39 @@ impl State {
...
-    pub(crate) fn new_synced() -> Self {
+    pub(crate) fn new_already_stored() -> Self {
         State {
             stored: true,
             ..Self::default()
         }
     }
 
+    /// Returns a new `State` with all fields set to `true` except for `sent_to_deploy_buffer` and
+    /// `sent_to_consensus_post_execution`.
```
Comment (rafal-ch) [inline node/src/types/block/meta_block/state.rs]: Nit:

```suggestion
    /// `consensus_notified`.
```

but, tbh, I'd rather remove the comment or rephrase it to something like "returns a new State which the block is expected to have after it has been synced"...

**Review by rafal-ch — APPROVED** [review]
LGTM with minor nits only :+1:

File: node/src/types/block/meta_block/state.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/types/block/meta_block/state.rs]: Done in [8eeb2f8](https://github.com/casper-network/casper-node/pull/3569/commits/8eeb2f8ba77159331b3a25a1bbfc423618df8f60).

**Comment by Fraser999** [comment]
bors merge
